google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/InstanceAdminClient.java

```diff
@@ -0,0 +1,454 @@
+   * }
+   * }
+   * </pre>
+   *
+   * @param instanceName
+   * @param appProfile
+   * @return
+   */
+  public AppProfile createAppProfile(InstanceName instanceName, AppProfile appProfile) {
+    return InstanceAdminRequests.convertToAppProfile(stub.createAppProfileCallable().call(
+        composeCreateAppProfileRequest(instanceName, appProfile)));
+  }
+
+  public AppProfile getAppProfile(AppProfileName appProfileName) {
+    return InstanceAdminRequests.convertToAppProfile(
+        stub.getAppProfileCallable().call(composeGetAppProfileRequest(appProfileName)));
+  }
+
+  public List<AppProfile> listAppProfiles(InstanceName instanceName) {
+    return convertToAppProfiles(
+        stub.listAppProfilesCallable().call(composeListAppProfilesRequest(instanceName)));
+  }
+
+  /**
+   * 
+   * <pre>
+   * {@code 
+   * try(InstanceAdminClient instanceAdmin = InstanceAdminClient.create()) {
+   *   AppProfile appProfileAny = instanceAdmin.getAppProfile(
+   *       AppProfileName.of("project", "instance", "profile"));
+   *   
+   *   AppProfile appProfileCluster = instanceAdmin.updateAppProfile(
+   *       AppProfile.ofUpdateAppProfile(appProfileAny)
+   *         .updateDescription("newDescription")
+   *         .updateRouteToCluster("clusterToUse", false))
+   *         .get();
+   * }
+   * }
+   * </pre>
+   * 
+   * @param updateAppProfile
+   * @return
+   */
+  public ApiFuture<AppProfile> updateAppProfile(UpdateAppProfile updateAppProfile) {
+    OperationFuture<com.google.bigtable.admin.v2.AppProfile, UpdateAppProfileMetadata> future =
+        stub.updateAppProfileOperationCallable()
+            .futureCall(composeUpdateAppProfileRequest(updateAppProfile));
+
+    return transfromOperationFuture(future,
+        new ApiFunction<com.google.bigtable.admin.v2.AppProfile, AppProfile>() {
+          @Override
+          public AppProfile apply(com.google.bigtable.admin.v2.AppProfile input) {
+            return InstanceAdminRequests.convertToAppProfile(input);
+          }
+        });
+  }
+
+  public void deleteAppProfile(AppProfileName appProfileName, boolean ignoreWarnings) {
+    stub.deleteAppProfileCallable()
+        .call(composeDeleteAppProfileRequest(appProfileName, ignoreWarnings));
+  }
+
+  public Policy getIamPolicy(InstanceName instanceName) {
+    return InstanceAdminRequests.convertToPolicy(
+        stub.getIamPolicyCallable().call(composeGetIamPolicyRequest(instanceName)));
+  }
+
+  public Policy setIamPolicy(InstanceName instanceName, Policy policy) {
+    return InstanceAdminRequests.convertToPolicy(
+        stub.setIamPolicyCallable().call(composeSetIamPolicyRequest(instanceName, policy)));
+  }
+
+  public List<String> testIamPermissions(InstanceName instanceName, List<String> permissions) {
+    return stub.testIamPermissionsCallable()
+               .call(composeTestIamPermissionsRequest(instanceName, permissions))
+               .getPermissionsList();
+  }
+
+  /** compose proto request helpers * */
```

Comment:
What is the split in responsibility between compose* and the toProto methods on the models? Any reason why this can't be collapsed into the models?